test/boilerplate_suggestors/boilerplate_utilities_test.dart

```diff
@@ -317,49 +272,101 @@ void main() {
       });
     });
 
-    group('isPublic() and getPublicExportLocations()', () {
-      setUpAll(() {
-        semverHelper = SemverHelper(jsonDecode(reportJson));
+    group('getSemverHelper() with isPublic() and getPublicExportLocations()',
+        () {
+      group('with --treat-all-components-as-private flag', () {
+        semverUtilitiesTestHelper(
+          path: 'test/boilerplate_suggestors/semver_report.json',
+          shouldTreatAllComponentsAsPrivate: true,
+          isValidFilePath: true,
+        );
       });
 
-      test('if props class is not in export list', () {
-        final input = '''
-          @Props()
-          class _\$FooProps extends UiProps{
-            String foo;
-            int bar;
-          }
-        ''';
-
-        CompilationUnit unit = parseString(content: input).unit;
-        expect(unit.declarations.whereType<ClassDeclaration>().length, 1);
-
-        unit.declarations.whereType<ClassDeclaration>().forEach((classNode) {
-          expect(semverHelper.getPublicExportLocations(classNode), isEmpty);
-          expect(isPublic(classNode), false);
+      group('json file does not exist', () {
+        semverUtilitiesTestHelper(
+          path: 'test/boilerplate_suggestors/does_not_exist.json',
+          shouldTreatAllComponentsAsPrivate: false,
+          isValidFilePath: false,
+        );
+
+        group('with --treat-all-components-as-private flag', () {
+          semverUtilitiesTestHelper(
+            path: 'test/boilerplate_suggestors/semver_report.json',
+            shouldTreatAllComponentsAsPrivate: true,
+            isValidFilePath: false,
+          );
         });
       });
 
-      test('if props class is in export list', () {
-        final input = '''
+      group('json file does exist', () {
+        semverUtilitiesTestHelper(
+          path: 'test/boilerplate_suggestors/semver_report.json',
+          shouldTreatAllComponentsAsPrivate: false,
+          isValidFilePath: true,
+        );
+      });
+    });
+  });
+}
+
+void semverUtilitiesTestHelper({
+  String path,
+  bool shouldTreatAllComponentsAsPrivate,
+  bool isValidFilePath,
```

Comment:
same #supernit as from the advanced_migrator test; you could add default parameters here and I feel like the invocations are a little cleaner, but either way!